internal/fsutil/create_test.go

```diff
@@ -209,15 +278,26 @@ func (s *S) TestCreate(c *C) {
 
 		c.Assert(err, IsNil)
 		c.Assert(testutil.TreeDump(dir), DeepEquals, test.result)
+
 		// [fsutil.Create] does not return information about parent directories
 		// created implicitly. We only check for the requested path.
-		entry.Path = strings.TrimPrefix(entry.Path, dir)
-		// Add the slashes that TreeDump adds to the path.
-		slashPath := "/" + test.options.Path
-		if test.options.Mode.IsDir() {
-			slashPath = slashPath + "/"
+		if entry.LinkType == fsutil.TypeHardLink {
+			// We should test hard link entries differently to ensure that it
+			// produces a hard link indeed.
+			pathInfo, err := os.Lstat(entry.Path)
+			c.Assert(err, IsNil)
+			linkInfo, err := os.Lstat(entry.Link)
+			c.Assert(err, IsNil)
+			os.SameFile(pathInfo, linkInfo)
```

Comment:
The comments `fsutil.Create] does not return information about parent directories created implicitly. We only check for the requested path.` explains the necessity of this test. Since the size and sha is not filled for a hard link's `fsEntry`, the only change we can apply here is to remove the `SameFile` test for hard links, but still keep it excluded from ooo`TreeDumpEntry` tests.